src/routes/orders.js

```diff
@@ -1,95 +1,90 @@
-  usersDataBase.map(user => {
-    if (user.id === order.user) {
-      getUserById = user;
-    }
-  });
-
-  const userDirPath = path.join(
-    __dirname,
-    "../",
-    "db/",
-    "users",
-    `/${getUserById.user}`,
-    "/orders"
-  );
-
-  fs.mkdir(userDirPath, err => {
-    if (err) {
-      return console.log(err);
-    }
-  });
-
-  let orderName = [];
-  order.products.map(item => {
-    let productSearch = productsDataBase.find(elem => elem.id === Number(item));
-
-    if (productSearch) {
-      orderName.push(productSearch.name);
-    }
-  });
-
-  const orderDirPath = path.join(`${userDirPath}`, `${orderName}.json`);
-
-  fs.writeFile(orderDirPath, JSON.stringify(order), err => {
-    if (err) {
-      return console.log(err);
-    }
-    console.log(`New order file "${orderName}.json" already created!`);
-  });
-};
-
-router.post("/*", (req, res) => {
-  let body = "";
-
-  req.on("data", data => {
-    body = body + data;
-    console.log("Incoming order data!");
-  });
+router.get("/:orderId", (req, res) => {
+  const id = req.params.orderId;
+  Order.findById(id)
+    .exec()
+    .then(item => {
+      console.log(item);
+      res.status(200).json({
+        status: "success",
+        order: item
+      });
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({ error: err });
+    });
+});
 
-  req.on("end", () => {
-    const orderList = JSON.parse(body);
-    const foundedProd = getProducts(orderList);
-    orderList.id = shortid.generate();
+router.get("/", (req, res) => {
+  Order.find()
+    .exec()
+    .then(items => {
+      console.log(items);
+      res.status(200).json(items);
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({ error: err });
+    });
+});
 
-    if (foundedProd.length !== 0) {
-      makeOrderDir(orderList);
-      res.setHeader("Content-Type", "application/json");
-      res.send(JSON.stringify({ status: "success", order: orderList }));
-      res.end();
-    } else {
-      res.setHeader("Content-Type", "application/json");
-      res.send(JSON.stringify({ status: "failed", order: null }));
-      res.end();
-    }
-  });
+router.post("/", jsonParser, (req, res) => {
+  if (checkOrder(req.body)) {
+    console.log("Validation complete!");
+    const order = new Order({
+      _id: new mongoose.Types.ObjectId(),
+      creator: req.body.creator,
+      productsList: req.body.productsList,
+      deliveryType: req.body.deliveryType,
+      deliveryAdress: req.body.deliveryAdress,
+      sumToPay: req.body.sumToPay,
+      status: req.body.status
+    });
+
+    order
+      .save()
+      .then(result => {
+        console.log(result);
+        res.status(201).json({
+          status: "success",
+          order: order
+        });
+      })
+      .catch(err => {
+        console.log(err);
+        res.status(500).json({ error: err });
+      });
+  } else {
+    console.log("Validation error!");
+    res
+      .status(500)
```

Comment:
Validation error should not end in 500 error. Validation error - it's not `Server Error`, but `Client Error` group of errors